ampd/src/stellar/verifier.rs

```diff
@@ -48,34 +38,21 @@ impl PartialEq<ContractEventBody> for VerifierSetConfirmation {
     fn eq(&self, event: &ContractEventBody) -> bool {
         let ContractEventBody::V0(body) = event;
 
-        if body.topics.len() != 1 {
+        if body.topics.len() != 3 {
             return false;
         }
 
-        let [symbol] = &body.topics[..] else {
+        let [symbol, _, rotate_signers_hash] = &body.topics[..] else {
             return false;
         };
 
         let expected_topic: ScVal =
             ScSymbol(StringM::from_str(TOPIC_ROTATED).expect("must convert str to ScSymbol"))
                 .into();
 
-        let rotated_signers = match &body.data {
-            ScVal::Vec(Some(data)) if data.len() == 1 => {
-                let [rotated_signers] = &data[..] else {
-                    return false;
-                };
-                rotated_signers.clone()
-            }
-            _ => return false,
-        };
-
-        WeightedSigners::try_from(&self.verifier_set)
-            .ok()
-            .and_then(|signers| ScVal::try_from(signers).ok())
-            .map_or(false, |signers: ScVal| {
-                symbol == &expected_topic && signers == rotated_signers
-            })
+        expected_topic == *symbol
+            && ScVal::Bytes(ScBytes(BytesM::try_from(self.verifier_set.hash()).unwrap()))
+                == *rotate_signers_hash
```

Comment:
similar to this test, add a golden test for WeightedSigners `hash()` as well to detect any changes. You can add a similar test in soroban repo with the same weighted signers values so they match
```
    #[test]
    fn signers_rotation_hash() {
```

Reply:
that's a good call. updated.